project/SparkBuild.scala

```diff
@@ -480,7 +480,6 @@ object SparkParallelTestGrouping {
     "org.apache.spark.sql.hive.thriftserver.SparkSQLEnvSuite",
     "org.apache.spark.sql.hive.thriftserver.ui.ThriftServerPageSuite",
     "org.apache.spark.sql.hive.thriftserver.ui.HiveThriftServer2ListenerSuite",
-    "org.apache.spark.sql.hive.thriftserver.ThriftServerWithSparkContextSuite",
```

Comment:
There are multiple test suites from thriftserver moduel here. Are the other suites OK to be executed in parallel？

Reply:
I verified it locally with all `SharedThriftServer`-like tests in a single JVM, which needed to remove `org.apache.spark.sql.hive.thriftserver.ThriftServerQueryTestSuite` here, and passed.

command:
```sbt
build/sbt "hive-thriftserver/test-only *ThriftServer*" -Phive-thriftserver
```

Reply:
Let's make sure we can pass tests when running within a single JVM. We can send a new PR to optimize it by running tests in parallel.

Reply:
@cloud-fan I meant the other hive-thriftserver suites in `testsWhichShouldRunInTheirOwnDedicatedJvm`. 
From line 479 to 482. I haven't looked into it.

Reply:
I checked the log again and found that the 3 of `SharedThriftServer`-like tests in https://github.com/apache/spark/pull/28797#discussion_r438632059 were executed sequentially in the single JVM.

https://www.scala-sbt.org/1.x/docs/Testing.html#Forking+tests

> By default, sbt runs all tasks in parallel and within the same JVM as sbt itself.

> The tests in a single group are run sequentially.

Not quite sure which rule above worked